armi/cli/clone.py

```diff
@@ -28,17 +28,16 @@ class CloneArmiRunCommandBatch(EntryPoint):
     settingsArgument = "required"
 
     def addOptions(self):
-        for settingName in self.cs.keys():
-            # verbosity and branchVerbosity already have command line options in the default parser
-            # adding them again would result in an error from argparse.
-            if settingName not in ["verbosity", "branchVerbosity"]:
-                self.createOptionFromSetting(settingName)
         self.parser.add_argument(
             "--additional-files",
             nargs="*",
             default=[],
             help="Additional files from the source directory to copy into the target directory",
         )
+        # somehow running `armi clone-batch -h` on the command line requires this to
+        # not be first?
+        for settingName in self.cs.keys():
+            self.createOptionFromSetting(settingName, suppressHelp=True)
```

Comment:
anyone have any clue why this works?

Reply:
I can reproduce that error:

```
  File "c:\users\jstilley\codes\repos\armi\armi\__main__.py",
    code = ArmiCLI().run()
...
    assert ' '.join(opt_parts) == opt_usage
AssertionError
```

I can verify that the code runs ALL the way through this method without any errors, and then fails on `parse_args` in `entryPoint.py`.

My take-away from this is that the `nargs="*"`  is problematic.  You can see it [here](https://docs.python.org/3/library/argparse.html#nargs), and they don't make any warning about using the glob `*`, but that seems like the issue to me.

Reply:
Ah ok that is somewhat helpful. I guess if you have `n` optional args you can't have an additional optional arg defined after that, or it would get processed with the first group of optional args...???

Reply:
Yeah, my guess is it's something like that.

It is either a bug in the `argparse` standard library, or just the logical result of saying "we will take ANY extra args" with `nargs="*"`.  Like, maybe that's just logically sound, but limiting.